nodestream/pipeline/extractors/queues/extractor.py

```diff
@@ -0,0 +1,88 @@
+import json
+from abc import ABC, abstractmethod
+from logging import getLogger
+from typing import Any, Iterable
+
+from ....model import JsonLikeDocument
+from ....pluggable import Pluggable
+from ....subclass_registry import SubclassRegistry
+from ...flush import Flush
+from ..extractor import Extractor
+
+QUEUE_CONNECTOR_SUBCLASS_REGISTRY = SubclassRegistry()
+QUEUE_OBJECT_FORMAT_SUBCLASS_REGISTRY = SubclassRegistry()
+
+
+@QUEUE_CONNECTOR_SUBCLASS_REGISTRY.connect_baseclass
+class QueueConnector(Pluggable, ABC):
+    entrypoint_name = "queue_connectors"
+
+    @abstractmethod
+    async def poll(self) -> Iterable[Any]:
+        raise NotImplementedError
+
+
+@QUEUE_OBJECT_FORMAT_SUBCLASS_REGISTRY.connect_baseclass
+class QueueRecordFormat(Pluggable, ABC):
+    entrypoint_name = "record_formats"
+
+    @abstractmethod
+    def parse(self, record: Any) -> JsonLikeDocument:
+        raise NotImplementedError
+
+
+class JsonStreamRecordFormat(QueueRecordFormat, alias="json"):
+    def parse(self, record: Any) -> JsonLikeDocument:
+        return json.loads(record)
+
+
+class QueueExtractor(Extractor):
+    """A QueueExtractor implements the standard behavior of polling data from a stream.
+
+    The QueueExtractor requires both a QueueConnector and a QueueRecordFormat to delegate
+    to for the actual polling implementation and parsing of the data, respectively.
+    """
+
+    @classmethod
+    def from_file_data(cls, connector: str, record_format: str, **connector_args):
```

Comment:
It is possible to add a test case for this? It looks like it may be tricky and that may have been why you omitted it in which case I am okay.

Reply:
Added